### PR TITLE
Create filter component

### PR DESCRIPTION
**Pull request by BenjaminParment**
-> Adds Pills reusable component
-> Adds optional onClick to svg props
-> Create basic filter display
-> Renders the ExactBatch filter in transactions

-> will write tests after first review

-> The css when the pills are in an active state needs to be reconfigured
-> A transparent border was added to the pill so that when in focus state, the pill stays in the same position is it

### CODE REVIEW

File: src/components/filter-display/FilterDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import TransactionExactFilter from "./transaction-exact-filter/TransactionExactFilter";
+import { connect } from "react-redux";
+
+const mapStateToProps = (state: RootState) => ({
+    exactBatch: state.transactionsFilter.exactBatch,
+});
+
+type ConnectedFilterDisplayProps = ReturnType<typeof mapStateToProps> & RouteComponentProps;
+
+class FilterDisplay extends React.Component<ConnectedFilterDisplayProps> {
```
Comment (j-n-white) [inline src/components/filter-display/FilterDisplay.tsx]: This name is a little misleading. I thought it was the component with all the fields to select the filters. How about `SelectedFilters`?

File: src/components/filter-display/FilterDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    render = () => (
+        <div className="filter-display-parent-container">
+            {this.props.location.pathname === TRANSACTIONS_PATH && this.props.exactBatch && (
```
Comment (j-n-white) [inline src/components/filter-display/FilterDisplay.tsx]: Why are you checking the path? Just only include this component on the correct page.

File: src/components/filter-display/filter-row/FilterRow.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import React from "react";
+
+interface FilterRowProps<T> {
+    filterTitle: string;
+    filterContent: T[];
+    removeFilter: (data: T) => void;
+    extractName: (data: T) => string;
+}
+
+export class FilterRow<T> extends React.Component<FilterRowProps<T>> {
```
Comment (j-n-white) [inline src/components/filter-display/filter-row/FilterRow.tsx]: Again `SelectedFilterRow` would be clearer

File: src/components/filter-display/filter-row/FilterRow.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    render = () => (
+        <div className="filter-row-parent-container">
```
Comment (j-n-white) [inline src/components/filter-display/filter-row/FilterRow.tsx]: Don't we want some logic to not show this if there is no filter content?

File: src/components/filter-display/transaction-exact-filter/TransactionExactFilter.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+type ConnectedTransactionExactFilterProps = ReturnType<typeof mapDispatchToProps> & TransactionExactFilterProps;
+
+class TransactionExactFilter extends React.Component<ConnectedTransactionExactFilterProps> {
+    render = () => (
+        <FilterRow
+            filterTitle={"Exact match:"}
+            filterContent={[this.props.exactBatch]}
+            removeFilter={this.props.removeExactBatchId}
+            extractName={(batch: ExactBatchFilter) => batch.name}
```
Comment (j-n-white) [inline src/components/filter-display/transaction-exact-filter/TransactionExactFilter.tsx]: Move this to a separate method so it doesn't create the function on each render

File: src/components/reusables/pills/Pills.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+export class Pills<T> extends React.Component<PillsProps<T>> {
+    private buttonClick = () => {
+        this.props.onClick(this.props.dataToRemove);
+    };
+
+    render = () => (
+        <div className="pill-parent-container">
+            <span className="content">{this.props.content}</span>
+            <div className="close-icon" tabIndex={0} aria-label={`Remove filter ${this.props.content}`}>
```
Comment (j-n-white) [inline src/components/reusables/pills/Pills.tsx]: If this is a button make it a button html element

File: src/components/submissions/Submissions.tsx [hunk 7]
```diff
@@ -49,6 +50,7 @@ export class Submissions extends React.Component<ConnectedSubmissionsPageProps>
             <div className="submissions-page-result-number-container">
                 <span className="result-number">{formatResultsCount(this.props.submissionsPage?.totalElements)}</span>
             </div>
+            <FilterDisplay />
```
Comment (j-n-white) [inline src/components/submissions/Submissions.tsx]: Why is this here?

File: src/components/submissions/Submissions.scss [hunk 8]
```diff
@@ -39,6 +39,10 @@ $SUBMISSIONS_MARGIN_BOTTOM: 58px;
         }
     }
 
+    .filter-display {
+        min-height: 50px;
+    }
+
```
Comment (j-n-white) [inline src/components/submissions/Submissions.scss]: This also should not be here

File: src/components/transactions/transactions-filter/transactionsFilter.ts [hunk 9]
```diff
@@ -1,6 +1,11 @@
 import { PagingFilter } from "components/paging/pagingFilter";
 
+export interface ExactBatchFilter {
```
Comment (j-n-white) [inline src/components/transactions/transactions-filter/transactionsFilter.ts]: This is likely to be the same for the other batch filters so call it `BatchFilter`

File: src/resources/svgs/svgProps.ts [hunk 10]
```diff
@@ -2,4 +2,5 @@ export interface SVGProps {
     height: number;
     width: number;
     className?: string;
+    onClick?: () => void;
```
Comment (j-n-white) [inline src/resources/svgs/svgProps.ts]: Do we need this? I think it would be better to wrap the svg in a button.

File: src/components/filter-display/filter-row/FilterRow.tsx (hunk 4 above)
Comment (BenjaminParment) [reply src/components/filter-display/filter-row/FilterRow.tsx]: should the logic be in this component or in the component above (transactions/submissions)?

Comment (j-n-white) [reply src/components/filter-display/filter-row/FilterRow.tsx]: I think it makes sense in the row component as it will be the same for each row

File: src/components/selected-filter/SelectedFilters.tsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import { connect } from "react-redux";
+
+const mapStateToProps = (state: RootState) => ({
+    exactBatch: state.transactionsFilter.exactBatch,
+});
+
+type ConnectedSelectedFiltersProps = ReturnType<typeof mapStateToProps> & RouteComponentProps;
+
+class SelectedFilters extends React.Component<ConnectedSelectedFiltersProps> {
+    render = () => <div className="filter-display-parent-container"></div>;
```
Comment (j-n-white) [inline src/components/selected-filter/SelectedFilters.tsx]: There is something missing here I think.

File: src/components/selected-filter/SelectedFilters.tsx [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+export default connect(mapStateToProps)(withRouter(SelectedFilters));
```
Comment (j-n-white) [inline src/components/selected-filter/SelectedFilters.tsx]: You don't need to connect this to the router any more.

File: src/components/selected-filter/transaction-exact-filter/TransactionExactFilter.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+type ConnectedTransactionExactFilterProps = ReturnType<typeof mapDispatchToProps> & TransactionExactFilterProps;
+
+class TransactionExactFilter extends React.Component<ConnectedTransactionExactFilterProps> {
+    private extractName = (batch: BatchFilter) => batch.name;
+
+    render = () => (
+        <SelectedFilterRow
+            filterTitle={"Exact match:"}
+            filterContent={[this.props.exactBatch]}
```
Comment (j-n-white) [inline src/components/selected-filter/transaction-exact-filter/TransactionExactFilter.tsx]: Exact batch could be null. This would then pass an array through with one item which would be null. This would however still have a length so the pill would still render. Probably best to put a null check here and then pass through an empty array

File: src/components/reusables/pills/Pills.tsx (hunk 6 above)
Comment (j-n-white) [reply src/components/reusables/pills/Pills.tsx]: Move the tab index and aria-label to the button. Do we even need that surrounding div any more?

File: src/components/reusables/pills/Pills.tsx [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+import { CloseIcon } from "resources/svgs/CloseIcon";
+import React from "react";
+
+interface PillsProps<T> {
+    content: string;
+    dataToRemove: T;
+    onClick: (data: T) => void;
+}
+
+export class Pills<T> extends React.Component<PillsProps<T>> {
```
Comment (j-n-white) [inline src/components/reusables/pills/Pills.tsx]: Sorry I didn't notice this before but this is a single `Pill` not `Pills`

File: src/components/transactions/Transactions.tsx [hunk 15]
```diff
@@ -54,6 +56,10 @@ export class Transactions extends React.Component<ConnectedTransactionsProps> {
                     {formatTransactionsNumber(this.props.transactionsPage?.totalElements)}
                 </span>
             </div>
+            {this.props.transactionsFilter.exactBatch && (
+                <TransactionExactFilter exactBatch={this.props.transactionsFilter.exactBatch} />
+            )}
+            <SelectedFilters />
```
Comment (j-n-white) [inline src/components/transactions/Transactions.tsx]: Surely the TransactionExactFilter should be inside the SelectedFilters component?

File: src/components/selected-filter/SelectedFilters.tsx (hunk 11 above)
Comment (BenjaminParment) [reply src/components/selected-filter/SelectedFilters.tsx]: I can add this.props.children (to render the exactTransactionFilter), apart from that I guess the rest will come when we implement more filters?

File: src/components/transactions/Transactions.tsx [hunk 16]
```diff
@@ -56,10 +56,11 @@ export class Transactions extends React.Component<ConnectedTransactionsProps> {
...
             </div>
-            {this.props.transactionsFilter.exactBatch && (
-                <TransactionExactFilter exactBatch={this.props.transactionsFilter.exactBatch} />
-            )}
-            <SelectedFilters />
+            <SelectedFilters>
+                {this.props.transactionsFilter.exactBatch && (
+                    <TransactionExactFilter exactBatch={this.props.transactionsFilter.exactBatch} />
+                )}
+            </SelectedFilters>
```
Comment (j-n-white) [inline src/components/transactions/Transactions.tsx]: This isn't quite what I meant. Given that Selected filters is only used for transactions then this logic should be in that class.
 so this component would just have 
```
            <SelectedTransactionsFilters />
```
selected transactions filters would have 
```
<div className="filter-display-parent-container">
    <TransactionExactFilter />
</div>
```

And the TransactionExactFilter is bound to redux
